fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MaterializedViewTransparentRewriteRule.java

```diff
@@ -329,35 +327,14 @@ private OptExpression getMvDefinedQueryPlan(OptExpression mvPlan,
                                                 MaterializationContext mvContext,
                                                 List<ColumnRefOperator> originalOutputColumns) {
         OptExpressionDuplicator duplicator = new OptExpressionDuplicator(mvContext);
-        OptExpression newMvQueryPlan = duplicator.duplicate(mvPlan);
-        newMvQueryPlan = MVPartitionPruner.resetSelectedPartitions(newMvQueryPlan, true);
+        OptExpression newMvQueryPlan = duplicator.duplicate(mvPlan, true, true);
 
         List<ColumnRefOperator> orgMvQueryOutputColumnRefs = mvContext.getMvOutputColumnRefs();
         List<ColumnRefOperator> newQueryOutputColumns = duplicator.getMappedColumns(orgMvQueryOutputColumnRefs);
         Map<ColumnRefOperator, ScalarOperator> newProjectionMap = Maps.newHashMap();
         for (int i = 0; i < originalOutputColumns.size(); i++) {
             newProjectionMap.put(originalOutputColumns.get(i), newQueryOutputColumns.get(i));
         }
-        return mergeProjection(newMvQueryPlan, newProjectionMap);
-    }
-
-    private OptExpression mergeProjection(OptExpression input, Map<ColumnRefOperator, ScalarOperator> newProjectionMap) {
```

Comment:
so mergeProjection is removed to Utils.java for the purpose of refactoring?

Reply:
yes.